chore(tools): change .eslintrc to be compatible with eslint 2.0

Akath19 opened the pull request:
ESLint 2.0 removed the rules "space-after-keywords" and "space-return-throw-case" and replaced them with "keyword-spacing", said rule controls spacing behaviour for spaces before and after keywords, .eslintrc has been changed to reflect that and now spaces are necessary before and after each keyword

BREAKING CHANGES: compatibility with ESLint < 2.0

akath19 commented:
Previous PR caused errors with spaces before some keywords.

gheoan commented on .eslintrc:
```diff
@@ -158,11 +158,10 @@
       "after": true
     }],
-    "space-after-keywords": 2,       // http://eslint.org/docs/rules/space-after-keywords
+    "keyword-spacing": [2, {"before": true, "after": true}] //http://eslint.org/docs/rules/keyword-spacing
```
@Akath19 Missing comma here :smile:  .

akath19 replied:
@Gheoan :angry: Don't you hate it when a comma breaks an entire system!!!

EisenbergEffect commented:
Would it be possible to update this in a way that it is backwards compatible with 1.x?

akath19 commented:
The rules were deprecated and aren't available on 2.X, so there are two ways to keep it backwards compatible:

1) Import the deprecated rules to a lib folder.
2) Keep a copy of .eslintrc for 1.X, one for 2.X and choosing the right one via NPM scripts.

I really don't think option 1 is a good idea because we would have overlapping rules (the deprecated ones and the current ones). I'm currently working on the NPM script version in order to select the proper file depending on the installed ESLint version.

akath19 commented:
I've updated the PR with the npm script that will select the appropriate .eslintrc file according to the globally installed version.

I also deleted the original .eslintrc in order to only select the correct one from the scripts folder, this makes the change compatible with both 1.x and 2.x

RWOverdijk commented:
He, was about to start working on this. Glad I checked if someone perhaps already did first. This PR has my  :+1:

Also, I think this is as BC as it gets.

EisenbergEffect commented:
Hmm. This seems it requires eslint to be globally installed, which probably isn't common for our community. Also, will the installation fail if it isn't installed? @bryanrsmith Can you review this? Any thoughts on how to handle this?

bryanrsmith commented:
Why is it desirable to maintain compat with 1.0? I think we should just update the eslint dependency.

EisenbergEffect commented:
It seems like it would silently break anyone who updated the aurelia tools. I guess what we need to do is up the minor version on aurelia tools so that it can't break anyone on an update? Then we need to update all our repos. Is that correct?

bryanrsmith commented:
Yep.
On Tue, Apr 26, 2016 at 8:03 AM Rob Eisenberg notifications@github.com
wrote:

> It seems like it would silently break anyone who updated the aurelia
> tools. I guess what we need to do is up the minor version on aurelia tools
> so that it can't break anyone on an update? Then we need to update all our
> repos. Is that correct?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/aurelia/tools/pull/28#issuecomment-214775001

EisenbergEffect commented:
@Akath19 Sorry to be such a pain. Would you mind reverting so that your PR only supports ESLint 2 and rebasing? 😄

akath19 commented:
@EisenbergEffect No problem, already updated
I'm closing this one in favor of #36, it's clean and also I accidentally broke the repo on this one :flushed:
